### PR TITLE
Create repro test for SharedTree bug

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
Found a `SharedTree` eventual consistency/sync bug, this repros it.

Repro steps
1. Create **Container A** with a `SharedTree`
2. Load **Container B** with a `SharedTree`
3. Schematize **A** and **B** with Schema
4. Summarize - **Summary 1**
5. Load **Container C** from **Summary 1**
6. Schematize **C** with **Schema**
7. The state **Container C** has does NOT sync with **Container A** and does NOT sync with **Container B**
8. Sending ops from **Container C** only updates **Container C** and not **A** and **B**

I verified that the ops were seen by all `SharedTree` instances
I verified that using a `SharedMap` had no issues in syncing 
I verified that the **Summary 1** generated by the summarizer seems to have the data (I set a value to 5) that I want.
I'm using the API surface outlined in the example in `examples/data-objects/inventory-app/src/inventoryList.ts`

### CODE REVIEW

File: packages/dds/migration-shim/src/test/sharedTreeBug.spec.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		// Get the views and root object to modify
+		const view1 = getNewTreeView(tree1);
+		const view2 = getNewTreeView(tree2);
+		const treeNode1 = view1.root as unknown as Typed<typeof someType>;
+		const treeNode2 = view2.root as unknown as Typed<typeof someType>;
+
+		await provider.ensureSynchronized();
+
+		// Modify the tree
+		treeNode1.quantity = testValue;
```
Comment (connorskees) [inline packages/dds/migration-shim/src/test/sharedTreeBug.spec.ts:126]: I think only the proxy-based API is allowed to set values using `=`?

```suggestion
		const treeNode1 = view1.root2(schema);
		const treeNode2 = view2.root2(schema);

		await provider.ensureSynchronized();

		// Modify the tree
		treeNode1.quantity = testValue;
```

**Comment by tyler-cai-microsoft** [comment]
I run into `0x77f` with the other change

File: packages/dds/migration-shim/src/test/sharedTreeBug.spec.ts (hunk 1 above)
Comment (tyler-cai-microsoft) [reply packages/dds/migration-shim/src/test/sharedTreeBug.spec.ts]: I run into 0x77f with the other change

**Comment by connorskees** [comment]
Yeah that's the error I was hitting last night after making a similar change. I still think using root2 is more correct, but there is probably some weird bug here.

**Comment by tyler-cai-microsoft** [comment]
> Yeah that's the error I was hitting last night after making a similar change. I still think using root2 is more correct, but there is probably some weird bug here.

I'm open to suggestions

**Comment by connorskees** [comment]
To be clear I think it looks like the bug is in shared-tree and not in the test itself

**Pull request closed without merging**